### PR TITLE
New design of threads on diff page

### PR DESCRIPTION
**Pull request by webdevelopland**
New design and functionality of threads on diff page.

### CODE REVIEW

File: tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/diff-threads/diff-threads.component.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+// The component implements diff threads on diff page.
+// How it looks: https://i.imgur.com/fYZn2qn.jpg
+@Component({
+  selector: 'diff-threads',
+  templateUrl: './diff-threads.component.html',
+  styleUrls: ['./diff-threads.component.scss'],
+  providers: [DiscussionService],
+})
+export class DiffThreadsComponent implements OnInit, OnChanges {
+  displayedColumns = ['discussions'];
```
Comment (MateMahinKhan) [inline tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/diff-threads/diff-threads.component.ts]: Hey bro, any reason this isn't type enforced

File: tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread/thread.component.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+  selector: 'cr-thread',
+  templateUrl: './thread.component.html',
+  styleUrls: ['./thread.component.scss'],
+})
+export class ThreadComponent implements OnInit, OnDestroy {
+  isCommentOpenMap: boolean[] = [];
+  isReply: boolean = false;
+  textarea: FormControl = new FormControl();
+  resolvedCheckbox: FormControl = new FormControl();
+  subscription = new Subscription();
```
Comment (MateMahinKhan) [inline tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread/thread.component.ts]: Let's put type enforcement here

Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread/thread.component.ts]: I think type enforcement is redundant with `new`.
We already tell anybody who review the code that type is `Subscription`. Code editor or IDE also know that it's `Subscription` even without new.
So let's delete type enforcement to `textarea` and `resolvedCheckbox`

Comment (webdevelopland) [reply tools/reviewer/webapp/src/app/dashboard/diff/diff-discussion/thread/thread.component.ts]: @MateMahinKhan Also, do you know a way to use typedef option of tslint, but to not force type with new?

**Pull request merged**